rosmon_core/src/monitor/node_monitor.cpp

```diff
@@ -172,13 +173,27 @@ void NodeMonitor::start()
 	if(running())
 		return;
 
-	if(m_launchNode->coredumpsEnabled() && g_coreIsRelative && m_processWorkingDirectory.empty())
+	if(m_launchNode->coredumpsEnabled() && g_coreIsRelative)
 	{
-		char tmpfile[256];
-		strncpy(tmpfile, "/tmp/rosmon-node-XXXXXX", sizeof(tmpfile));
-		tmpfile[sizeof(tmpfile)-1] = 0;
-		m_processWorkingDirectory = mkdtemp(tmpfile);
-		m_processWorkingDirectoryCreated = true;
+		if (!m_processWorkingDirectory.empty()) 
+		{
+			std::string dir = m_processWorkingDirectory + "/core_dumps";
+			if (chdir(dir.c_str()) == 0 || mkdir(dir.c_str(), 0777) == 0) 
+			{
+				m_processWorkingDirectoryCreated = true;
+			} 
+			else 
+			{
+				perror("Could not create rosmon/core_dumps directory");
```

Comment:
Please replace perror lines with the log formatting type that is used in this repo, namely `logTyped(LogEvent::Type::Error, ..)`

Reply:
The same for all other places you are using perror